File: classify-server/src/main/java/io/ermdev/alice/repository/StudentRepository.java
package io.ermdev.alice.repository;

import io.ermdev.alice.entity.Student;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

public interface StudentRepository extends JpaRepository<Student, Long> {

    @Query("from Student where id=:studentId")
    Student findById(@Param("studentId") Long studentId);
}

File: classify-spring-boot-webservices/src/main/kotlin/io/ermdev/classify/data/repository/StudentRepository.kt
package io.ermdev.classify.data.repository

import io.ermdev.classify.data.entity.Lesson
import io.ermdev.classify.data.entity.Student
import io.ermdev.classify.data.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface StudentRepository : JpaRepository<Student, Long> {

    @Query("from User as u join u.student as s where s.id=:studentId")
    fun findUser(@Param("studentId") studentId: Long): User?

    @Query("from Lesson as a join a.students as b where b.id=:studentId")
    fun findLessons(@Param("studentId") studentId: Long): List<Lesson>

    @Query("from Lesson as a join a.students as b where b.id=:studentId and a.id=:lessonId")
    fun findLesson(@Param("studentId") studentId: Long,
                   @Param("lessonId") lessonId: Long): Lesson?
}